src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -610,4 +613,147 @@ public String getDuplicateNameCandidate(UUID directoryUuid, String elementName,
         }
         return nameCandidate(elementName, i);
     }
+
+    public void restoreElements(List<UUID> elementsUuid, UUID parentUuid, String userId) {
+        // Get all updatable entities
+        List<DirectoryElementEntity> notUpdatableEntities = new ArrayList<>();
```

Comment:
This arraylist has no usage in this method, what is its purpose ?

Reply:
it is used to check if there is elements are not allowed to be updated, but i forget to throw exception when it is not empty. i fixed it

Reply:
It is supposed to throw exception when it is not empty. i fixed it